grails-app/utils/org/broadinstitute/orsp/utils/DulPdfParser.groovy

```diff
@@ -0,0 +1,228 @@
+                    break
+                case DataUseLetterFields.DATA_USE_CONSENT.abbreviation:
+                    field.setValue(dulInfoObj[DataUseLetterFields.DATA_USE_CONSENT.abbreviation].toString().isEmpty() ? 'Off' : dulInfoObj[DataUseLetterFields.DATA_USE_CONSENT.abbreviation].toString())
+                    break
+                case DataUseLetterFields.DATA_DEPOSITION_DESCRIBED.abbreviation:
+                    field.setValue(dulInfoObj[DataUseLetterFields.DATA_DEPOSITION_DESCRIBED.abbreviation].toString().isEmpty() ? 'Off' : dulInfoObj[DataUseLetterFields.DATA_DEPOSITION_DESCRIBED.abbreviation].toString())
+                    break
+                case DataUseLetterFields.REPOSITORY_TYPE.abbreviation:
+                    field.setValue(dulInfoObj[DataUseLetterFields.REPOSITORY_TYPE.abbreviation].toString().isEmpty() ? 'Off' : dulInfoObj[DataUseLetterFields.REPOSITORY_TYPE.abbreviation].toString())
+                    break
+                case DataUseLetterFields.GSR_AVAILABILITY.abbreviation:
+                    field.setValue(dulInfoObj[DataUseLetterFields.GSR_AVAILABILITY.abbreviation].toString().isEmpty() ? 'Off' : dulInfoObj[DataUseLetterFields.GSR_AVAILABILITY.abbreviation].toString())
+                    break
+                case DataUseLetterFields.GCR_AVAILABILITY_SPECIFY.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.GCR_AVAILABILITY_SPECIFY.abbreviation] as String, dul.uid))
+                    break
+                case DataUseLetterFields.PRINTED_NAME.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.PRINTED_NAME.abbreviation] as String, dul.uid))
+                    break
+                case DataUseLetterFields.SIGNATURE.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.SIGNATURE.abbreviation] as String, dul.uid))
+                    break
+                case DataUseLetterFields.POSITION.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.POSITION.abbreviation] as String, dul.uid))
+                    break
+                case DataUseLetterFields.INSTITUTION.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.INSTITUTION.abbreviation] as String, dul.uid))
+                    break
+                case DataUseLetterFields.SUBMITTED_DATE.abbreviation:
+                    field.setValue(getDefaultValue(dulInfoObj[DataUseLetterFields.SUBMITTED_DATE.abbreviation] as String, dul.uid))
+                    break
+            }
+            field.setReadOnly(true)
+
+        }
+        acroForm
+    }
+
+
+    private static String getDefaultValue(String value, String uid){
+        StringUtils.isEmpty(value) ? "--" : filterNonPrintableValue(value, uid)
```

Comment:
Thank you for `filterNonPrintableValue`!

Reply:
Thank you for the reminder earlier!